src/whisper.cpp

```diff
@@ -4346,6 +4346,48 @@ const char * whisper_print_system_info(void) {
     return s.c_str();
 }
 
+// whisper_get_system_info_json
+// Returns system info as json, useful for ports
+// NOTE : While testing features->value always returned a numeric so it is currently unquoted
+//        If strings are likely then something slightly more type-aware will be required
+const char * whisper_get_system_info_json(void) {
+    static std::string s;
+
+    whisper_load_backends();
+
+    s  = "{";
+    s += "\"WHISPER\":{";
+    s += "\"COREML\":"    + std::to_string(whisper_has_coreml())     + ",";
+    s += "\"OPENVINO\":"  + std::to_string(whisper_has_openvino())   + "}";
+
+    for (size_t i = 0; i < ggml_backend_reg_count(); i++) {
+        auto * reg = ggml_backend_reg_get(i);
+        auto * get_features_fn = (ggml_backend_get_features_t) ggml_backend_reg_get_proc_address(reg, "ggml_backend_get_features");
+        if (get_features_fn) {
+            ggml_backend_feature * features = get_features_fn(reg);
+            s += ",\"";
+            s += ggml_backend_reg_name(reg);
+            s += "\":{";
+			auto first = true;
```

Comment:
Nit: same here with tab and a few of the lines below as well.

Reply:
Fixed the tabs. Also added two helpful state-related functions.
Moved everything to end of source files to make it easy to play with.
Also changed JSON to always return string values leaving it up to the caller to handle typing. This ensures that the returned JSON is always valid.

Reply:
Thanks!